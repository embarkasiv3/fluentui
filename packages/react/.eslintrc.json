{
  // react-internal has some overrides we need to mirror
  "extends": ["../react-internal/.eslintrc"],
  "root": true
}
