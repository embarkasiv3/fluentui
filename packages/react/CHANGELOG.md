# Change Log - @fluentui/react

This log was last generated on Thu, 19 Mar 2020 17:52:34 GMT and should not be manually modified.
