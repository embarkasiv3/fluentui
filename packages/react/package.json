{
  "name": "@fluentui/react",
  "version": "8.0.0",
  "description": "Reusable React components for building web experiences.",
  "main": "lib-commonjs/index.js",
  "module": "lib/index.js",
  "typings": "lib/index.d.ts",
  "sideEffects": [
    "lib/version.js"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/microsoft/fluentui"
  },
  "license": "MIT",
  "scripts": {
    "build": "just-scripts build",
    "bundle": "just-scripts bundle",
    "clean": "just-scripts clean",
    "code-style": "just-scripts code-style",
    "codepen": "node ../../scripts/local-codepen.js",
    "just": "just-scripts",
    "lint": "just-scripts lint",
    "start": "just-scripts dev:storybook",
    "start:legacy": "yarn workspace @fluentui/public-docsite-resources start",
    "start-test": "just-scripts jest-watch",
    "test": "just-scripts test",
    "update-snapshots": "just-scripts jest -u",
    "mf": "just-scripts mf"
  },
  "devDependencies": {
    "@fluentui/eslint-plugin": "^1.0.0",
    "@types/chalk": "^2.0.0",
    "@types/enzyme": "3.10.3",
    "@types/enzyme-adapter-react-16": "1.0.3",
    "@types/glob": "^7.1.1",
    "@types/node": "^10.3.2",
    "@types/react": "16.9.42",
    "@types/react-dom": "16.9.10",
    "@types/react-test-renderer": "^16.0.0",
    "@types/webpack-env": "1.16.0",
    "@fluentui/scripts": "^1.0.0",
    "@fluentui/example-data": "^8.0.0",
    "@fluentui/jest-serializer-merge-styles": "^8.0.0",
    "@fluentui/test-utilities": "^8.0.0",
    "@fluentui/webpack-utilities": "^8.0.0",
    "@fluentui/react-conformance": "^0.2.1",
    "chalk": "^2.1.0",
    "enzyme": "~3.10.0",
    "enzyme-adapter-react-16": "^1.15.0",
    "glob": "^7.1.2",
    "jest-snapshot": "~24.9.0",
    "office-ui-fabric-core": "^11.0.0",
    "react": "16.8.6",
    "react-app-polyfill": "~1.0.1",
    "react-dom": "16.8.6",
    "react-test-renderer": "^16.3.0"
  },
  "dependencies": {
    "@fluentui/date-time-utilities": "^8.0.0",
    "@fluentui/foundation-legacy": "^8.0.0",
    "@fluentui/font-icons-mdl2": "^8.0.0",
    "@fluentui/merge-styles": "^8.0.0",
    "@fluentui/react-focus": "^8.0.0",
    "@fluentui/react-hooks": "^8.0.0",
    "@fluentui/react-shared-contexts": "^1.0.0-beta.9",
    "@fluentui/react-window-provider": "^2.0.0",
    "@fluentui/set-version": "^8.0.0",
    "@fluentui/style-utilities": "^8.0.0",
    "@fluentui/theme": "^2.0.0",
    "@fluentui/utilities": "^8.0.0",
    "@microsoft/load-themed-styles": "^1.10.26",
    "tslib": "^1.10.0"
  },
  "peerDependencies": {
    "@types/react": ">=16.8.0 <18.0.0",
    "@types/react-dom": ">=16.8.0 <18.0.0",
    "react": ">=16.8.0 <18.0.0",
    "react-dom": ">=16.8.0 <18.0.0"
  }
}
