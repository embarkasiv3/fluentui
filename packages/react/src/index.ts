import './version';
export * from '@fluentui/react-internal';
export * from './Breadcrumb';
export * from './Checkbox';
export * from './ComboBox';
export * from './DetailsList';
export * from './DocumentCard';
export * from './Dropdown';
export * from './GroupedList';
export * from './Link';
export * from './Pivot';
export * from './ShimmeredDetailsList';
export * from './Slider';
export * from './Toggle';
