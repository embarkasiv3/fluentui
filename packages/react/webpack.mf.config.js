const path = require('path');
const fs = require('fs');
const webpack = require('webpack');
const getResolveAlias = require('@fluentui/scripts/webpack/getResolveAlias');
const { createServeConfig } = require('@fluentui/scripts/webpack/webpack-resources');
const { webpackMerge } = require('just-scripts');

const shared = {
  react: { singleton: true },
  'react-dom': { singleton: true },
};

const rootComponents = fs
  .readdirSync(path.join(__dirname, 'src'))
  .filter(filename => filename[0].toUpperCase() === filename[0])
  .map(filename => filename.replace(/\.tsx?/, ''));
const rootComponentsExposes = {};
for (const component of rootComponents) {
  rootComponentsExposes[`./lib/${component}`] = `@fluentui/react/src/${component}`;
}

const myConfig = {
  output: {},
  entry: {},
  mode: 'development',
  devtool: 'cheap-module-source-map',
  optimization: {
    concatenateModules: false,
  },
  module: {
    rules: [],
  },
  plugins: [
    new webpack.container.ModuleFederationPlugin({
      name: 'fluentuiReact',
      filename: 'remoteEntry.js',
      exposes: {
        ...rootComponentsExposes,
        './lib/compat/Button': '@fluentui/react/src/compat/Button',
        '.': '@fluentui/react/src/index',
        './lib/index': '@fluentui/react/src/index',
      },
      shared,
    }),
  ],
  resolve: {
    alias: getResolveAlias(),
  },
  devServer: {
    port: 2345,
    contentBase: 'dist',
  },
};

const metaConfig = webpackMerge.merge(createServeConfig(), myConfig);

module.exports = metaConfig;
