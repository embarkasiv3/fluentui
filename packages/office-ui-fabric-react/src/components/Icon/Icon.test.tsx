import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { Icon } from './index';

describe('Icon', () => {
  it('renders Icon correctly', () => {
    const component = renderer.create(<Icon iconName='CompassNW' />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Icon correctly using iconName', () => {
    const component = renderer.create(<Icon iconName='Upload' />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});