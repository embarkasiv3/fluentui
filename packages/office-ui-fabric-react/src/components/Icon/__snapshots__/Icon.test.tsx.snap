// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders Icon correctly 1`] = `
<i
  aria-hidden={true}
  className=

      {
        -moz-osx-font-smoothing: grayscale;
        -webkit-font-smoothing: antialiased;
        display: inline-block;
        font-family: "FabricMDL2Icons-2";
        font-style: normal;
        font-weight: normal;
        speak: none;
      }
  data-icon-name="CompassNW"
  role="presentation"
>
  
</i>
`;

exports[`Icon renders Icon correctly using iconName 1`] = `
<i
  aria-hidden={true}
  className=

      {
        -moz-osx-font-smoothing: grayscale;
        -webkit-font-smoothing: antialiased;
        display: inline-block;
        font-family: "FabricMDL2Icons";
        font-style: normal;
        font-weight: normal;
        speak: none;
      }
  data-icon-name="Upload"
  role="presentation"
>
  
</i>
`;
